application/usecases/selectRoleDropdownUseCase.ts

```diff
@@ -0,0 +1,70 @@
+import { ActionRowBuilder, ButtonBuilder, ButtonStyle, CacheType, StringSelectMenuInteraction } from "discord.js";
+import AREA_ROLES_MAP from "../../assets/consts/areaRolesMap";
+import EXTRA_AREA_ROLES_MAP from "../../assets/consts/extraAreaRolesMap";
+import LANGUAGE_ROLES_MAP from "../../assets/consts/languageRolesMap";
+import { CustomMessage } from "../../domain/interface/customMessage.interface";
+import ChatService from "../../domain/service/chatService";
+
+export default class SelectRoleDropdownUseCase {
+  private chatService: ChatService;
+
+  constructor({ chatService }: { chatService: ChatService }) {
+    this.chatService = chatService;
+  }
+
+  async execute(interaction: StringSelectMenuInteraction<CacheType>) {
+    try {
+      if (!interaction.guildId) return;
+
+      const selected = interaction.values[0];
+
+      const allowedRoles = { ...AREA_ROLES_MAP, ...LANGUAGE_ROLES_MAP, ...EXTRA_AREA_ROLES_MAP };
+      const selectedRoleMap = allowedRoles ? allowedRoles[selected] : undefined;
+
+      if (!selectedRoleMap) return;
+
+      selectedRoleMap.id = await this.chatService.getRoleIdByName(interaction.guildId, selectedRoleMap.name);
+
+      if (!selectedRoleMap.id) return;
+
+      const hasRole = await this.chatService.isUserWithRoleName(interaction.guildId, interaction.user.id, [
+        selectedRoleMap.name,
+      ]);
+
+      if (hasRole) {
+        this.chatService.sendInteractionReply(
+          interaction,
+          SelectRoleDropdownUseCase.confirmRemoveRoleMessage(selectedRoleMap.id, selectedRoleMap.name)
```

Comment:
Penso que possas tirar a função de estática e chamar a própria instância?

```suggestion
          this.sendMessageConfirmRemoveRole(selectedRoleMap.id, selectedRoleMap.name)
```

Reply:
A função estática serviu para ajudar a construir a mesagem para os testes